src/main/xml-parsers.js

```diff
@@ -279,42 +280,101 @@ export function parseCompleteMultipart(xml) {
+  const parseCommonPrefixesEntity = responseEntity => {
+    if(responseEntity){
+      toArray(responseEntity).forEach((commonPrefix) => {
+        const prefix = toArray(commonPrefix.Prefix)[0]
+        result.objects.push({prefix, size: 0})
+      })
+    }
   }
-  xmlobj = xmlobj.ListBucketResult
-  if (xmlobj.IsTruncated) result.isTruncated = xmlobj.IsTruncated
-  if (xmlobj.Contents) {
-    toArray(xmlobj.Contents).forEach(content => {
-      var name = toArray(content.Key)[0]
-      var lastModified = new Date(toArray(content.LastModified)[0])
-      var etag = toArray(content.ETag)[0].replace(/^"/g, '').replace(/"$/g, '')
-        .replace(/^&quot;/g, '').replace(/&quot;$/g, '')
-        .replace(/^&#34;/g, '').replace(/&#34;$/g, '')
-      var size = content.Size
-      result.objects.push({name, lastModified, etag, size})
-      nextMarker = name
-    })
+
+  const listBucketResult = xmlobj.ListBucketResult
+  const listVersionsResult=xmlobj.ListVersionsResult
+
+  if(listBucketResult){
+    if ( listBucketResult.IsTruncated) {
+      isTruncated = listBucketResult.IsTruncated
+    }
+    if (listBucketResult.Contents) {
+      toArray(listBucketResult.Contents).forEach(content => {
+        const name = toArray(content.Key)[0]
+        const lastModified = new Date(toArray(content.LastModified)[0])
+        const etag = sanitizeETag(toArray(content.ETag)[0])
+        const size = content.Size
+        result.objects.push({name, lastModified, etag, size})
+      })
+    }
+    
+    if( listBucketResult.NextMarker){
+      nextMarker = listBucketResult.NextMarker
+    }
+    parseCommonPrefixesEntity(listBucketResult.CommonPrefixes)
   }
+    
+  if(listVersionsResult){
+    if(listVersionsResult.IsTruncated){
+      isTruncated = listVersionsResult.IsTruncated
+    }
 
-  if (xmlobj.CommonPrefixes) {
-    toArray(xmlobj.CommonPrefixes).forEach(commonPrefix => {
-      var prefix = toArray(commonPrefix.Prefix)[0]
-      var size = 0
-      result.objects.push({prefix, size})
-    })
+    if (listVersionsResult.Version) {
+      toArray(listVersionsResult.Version).forEach(content => {
+        result.objects.push(formatObjInfo(content))
+      })
+    }
+    if (listVersionsResult.DeleteMarker) {
+      toArray(listVersionsResult.DeleteMarker).forEach(content => {
+        result.objects.push(formatObjInfo(content))
+      })
+    }
+
+    if (listVersionsResult.NextKeyMarker) {
+      nextVersionKeyMarker = listVersionsResult.NextKeyMarker
+    }
+    if (listVersionsResult.NextVersionIdMarker) {
+      result.versionIdMarker = listVersionsResult.NextVersionIdMarker
+    }
+    parseCommonPrefixesEntity(listVersionsResult.CommonPrefixes)
   }
-  if (result.isTruncated) {
-    result.nextMarker = xmlobj.NextMarker ? toArray(xmlobj.NextMarker)[0]: nextMarker
+
+  if (isTruncated) {
+    result.isTruncated= isTruncated
+    result.nextMarker = nextVersionKeyMarker || nextMarker
```

Comment:
I guess it doesn't matter here and the code works, but for the sake of correctness, this code looks not quite right, if you don't see the caller side.
So, we should set `result.isTruncated` no matter what the value of `isTruncated` is.

```suggestion
  result.isTruncated= isTruncated
  if (isTruncated) {
    result.nextMarker = nextVersionKeyMarker || nextMarker
```